src/snowflake/cli/api/utils/dict_utils.py

```
@@ -0,0 +1,53 @@
+from __future__ import annotations
+
+
+def deep_merge_dicts(original_values: dict, override_values: dict):
+    """
+    Takes 2 dictionaries as input: original and override.
+
+    For every key in the override dictionary, override the same key
+    in the original dictionary, or create a new one if it doesn't exist.
+
+    If the override value and the original value are both dictionaries,
+    instead of overriding, recursively call this function to merge the keys of the sub-dictionaries.
+    """
+    if not isinstance(override_values, dict) or not isinstance(original_values, dict):
+        return
```

Raise a `ValueError` instead of returning immediately.